src/python/sdk/econia_sdk/view/market.py

```diff
@@ -320,6 +320,150 @@ def get_price_levels_all(view: EconiaViewer, market_id: int) -> dict:
     return get_price_levels(view, market_id)
 
 
+def get_open_orders_paginated(
+    viewer: EconiaViewer,
+    market_id: int,
+    ask_page_size: int,
+    bid_page_size: int,
+    next_ask_idx: int,
+    next_bid_idx: int,
+    ledger_version: int,
+):
+    returns = viewer.get_returns(
+        "market",
+        "get_open_orders_paginated",
+        [],
+        [
+            str(market_id),
+            str(ask_page_size),
+            str(bid_page_size),
+            str(next_ask_idx),
+            str(next_bid_idx),
+        ],
+        ledger_version=ledger_version,
+    )
+    return returns
+
+
+def get_open_orders_with_pagination(
+    viewer: EconiaViewer, market_id: int, max_levels: int = 100, page_size: int = 100
+) -> dict:
+    info = viewer.aptos_client.info()
+    ledger_version = int(info["ledger_version"])
+
+    next_ask_idx = "0"
+    next_bid_idx = "0"
+    bid_page_size = ask_page_size = page_size
+
+    asks = []
+    bids = []
+
+    keep_looping = True
+    while keep_looping:
+        returns = get_open_orders_paginated(
+            viewer,
+            market_id,
+            min(ask_page_size, max(max_levels - len(asks), 0)),
+            min(bid_page_size, max(max_levels - len(bids), 0)),
+            int(next_ask_idx),
+            int(next_bid_idx),
+            ledger_version,
+        )
+
+        value = returns[0]
+        next_ask_idx = returns[1]
+        next_bid_idx = returns[2]
+
+        for ask in value["asks"]:
+            asks.append(_convert_open_order_value(ask))
+        for bid in value["bids"]:
+            bids.append(_convert_open_order_value(bid))
+
+        if next_ask_idx == "0" and next_bid_idx == "0":
+            keep_looping = False
+        elif len(asks) >= max_levels or len(bids) >= max_levels:
+            keep_looping = False
+        elif next_ask_idx == "0" or next_bid_idx == "0":
+            if next_ask_idx == "0":
+                ask_page_size = 0
+            else:
+                bid_page_size = 0
```

Comment:
This will return early if too many of one side are indexed but not enough of the other, which disregard's the user's specific request

Instead, how about deleting the second condition (first `elif`), then moving the first condition (`if`) to come second?

Reply:
If we remove the second condition, then we won't stop until both sides are _exhausted_ which is not what we want to do. Even if we did, the first condition must remain the first because the `or` in the third condition would consume the `and` case.

Right now this logic is correct (I think) because both sides have equal max sizes. If they're different, as we're doing now, there needs to be some more logic. Will work on it.

Reply:
Handled in most recent revision